microprofile.jdt/com.redhat.microprofile.jdt.core/src/main/java/com/redhat/microprofile/jdt/internal/config/java/MicroProfileConfigHoverParticipant.java

```diff
@@ -100,43 +106,101 @@ public Hover collectHover(JavaHoverContext context, IProgressMonitor monitor) th
+		return new Hover(documentation, propertyKeyRange);
 	}
-
+	
+	public static Set<String> getMatchingKeys(String keyToMatch, JDTMicroProfileProject project) {
+		Set<String> result = new HashSet<String>();
+		Set<String> allKeys = project.getPropertyKeys();
+		for (String key: allKeys) {
+			if (key.endsWith(keyToMatch)) {
+				result.add(key);
+			}
+		}
+		return result;
+	}
+	
+	/**
+	 * Returns documentation about the provided keys in <code>propertyKeys<code>
+	 * 
+	 * @param propertyKeys      the property keys to create documentation for
+	 * @param project           the project
+	 * @param markdownSupported true if documentation must be formatted as markdown and
+	 *                          false otherwise
+	 * @param insertSpacing     true if spacing should be inserted around the equals
+	 *                          sign and false otherwise
+	 * @return documentation about the provided keys in <code>propertyKeys<code>
+	 * @throws JavaModelException
+	 */
+	public static MarkupContent getDocumentation(Set<String> propertyKeys, JDTMicroProfileProject project,
+			boolean markdownSupported, boolean insertSpacing) throws JavaModelException {
+		StringBuilder content = new StringBuilder();
+		
+		List<String> sortedKeys = propertyKeys.stream().sorted((o1, o2)->{
+			if (o1.charAt(0) != '%') return -1;
+			if (o2.charAt(0) != '%') return 1;
+			return o1.compareTo(o2);
+		}).collect(Collectors.toList());
+		for (String key: sortedKeys) {
+			String value = project.getProperty(key, null);
+			if (value != null) {
+				buildDocumentation(key, value, markdownSupported, insertSpacing, content);
+			}
+		}
+		
+		return new MarkupContent(markdownSupported ? MarkupKind.MARKDOWN : MarkupKind.PLAINTEXT, content.toString());
+		
+	}
+	
 	/**
 	 * Returns documentation about the provided <code>propertyKey</code>'s value,
 	 * <code>propertyValue</code>
 	 * 
-	 * @param propertyKey   the property key
-	 * @param propertyValue the property key's value
-	 * @param markdown      true if documentation must be formatted as markdown and
-	 *                      false otherwise
-	 * @param insertSpacing true if spacing should be inserted around the equals
-	 *                      sign and false otherwise
+	 * @param propertyKey       the property key
+	 * @param propertyValue     the property key's value
+	 * @param markdownSupported true if documentation must be formatted as markdown and
+	 *                          false otherwise
+	 * @param insertSpacing     true if spacing should be inserted around the equals
+	 *                          sign and false otherwise
 	 * @return
 	 */
 	public static MarkupContent getDocumentation(String propertyKey, String propertyValue,
-			DocumentFormat documentFormat, boolean insertSpacing) {
-		boolean markdown = DocumentFormat.Markdown.equals(documentFormat);
+			boolean markdownSupported, boolean insertSpacing) {

```

Comment:
Please keep DocumentFormat documentFormat parameter. We will able to manage another document format  with this signature.